pkg/operatorsource/handler_test.go

```diff
@@ -55,12 +65,16 @@ func TestHandle_PhaseHasChanged_UpdateExpected(t *testing.T) {
 	// We expect the transitioner to indicate that the object has changed and needs update.
 	transitioner.EXPECT().TransitionInto(&opsrcOut.Status.CurrentPhase, nextPhaseExpected).Return(true).Times(1)
 
-	// We expect the object to be updated successfully.
-	kubeclient.EXPECT().Update(ctx, opsrcOut).Return(nil).Times(1)
-
 	errGot := handler.Handle(ctx, opsrcIn)
 
 	assert.NoError(t, errGot)
+
+	// We expect the object to be updated successfully.
+	namespacedName := types.NamespacedName{Name: "foo", Namespace: "marketplace"}
+	opsrcGot := &v1alpha1.OperatorSource{}
+
+	assert.NoError(t, fakeclient.Get(ctx, namespacedName, opsrcGot))
```

Comment:
Same question as above. What's the point of asserting there isn't an error for fakeclient.Get()? Isn't that implied in the next assertion? What you are trying to say is "The operatorsource we are looking for really does exist", but I think we should only assert about errors on our own code. If this throws an error, does that necessarily mean that there was nothing found for that `namespacedName`?

Reply:
That's an interesting point. So what you're saying is, if `fakeclient.Get` does throw an error for some reason but `opsrcGot` matches `opsrcOut` then the test still passes since we're testing `handler` not the fake client. Is that right?